contracts/base/RegistryAdapter.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+import { IERC7484 } from "../interfaces/IERC7484.sol";
+
+/**
+ * IERC7484 Registry adapter.
+ * this feature is opt-in. The smart account owner can choose to use the registry and which
+ * attesters to trust
+ */
+abstract contract RegistryAdapter {
+    event ERC7484RegistryConfigured(IERC7484 indexed registry);
+
+    IERC7484 registry;
```

Comment:
```suggestion
    IERC7484 public registry;
```

Reply:
can we also rename to moduleRegistry and add ERC7484 in natspecs

Reply:
yeah I havent added natspec yet but happy to add it